buzsaki_lab_to_nwb/original_code/convert_yuta.m

```diff
@@ -1,162 +1,301 @@
+    'file_create_date', datestr(now, 'yyyy-mm-dd hh:MM:SS'), ...
+    'identifier', fname, ...
+    'session_start_time', session_start_time, ...
+    'general_experimenter', 'Yuta Senzai', ...
+    'general_session_id', sessionInfo.session.name((strfind(sessionInfo.session.name,'-')+1):end), ...
+    'general_institution', 'NYU');
 
-source = fname;
-file = nwbfile;
-file.file_create_date = {datestr(now, 'yyyy-mm-dd hh:MM:SS')};
-file.identifier = {fname};
-file.session_description = {fname};
-file.session_start_time = {sessionInfo.Date};
+
+%%
+nwb.general_subject = types.core.Subject( ...
+    'subject_id', '41', ...
+    'description', 'Yuta mouse 41', ...
+    'genotype', '', ... % unknown
+    'sex', 'U', ...
+    'species', 'Mus musculus');
 
 
 %% animal position
-whl_path = '/Users/bendichter/Desktop/Buzsaki/SenzaiBuzsaki2017/YutaMouse41-150903/YutaMouse41-150903.whl';
+%whl_path = '/Users/bendichter/Desktop/Buzsaki/SenzaiBuzsaki2017/YutaMouse41-150903/YutaMouse41-150903.whl';
+whl_path = 'D:\BuzsakiData\SenzaiY\YutaMouse41\YutaMouse41-150903\YutaMouse41-150903.whl';
 aa = dlmread(whl_path);
-fs = 1250./32.;
+fs = samplingRate / 32; % ? Cody: what is this for and what are the units?
+% Cody: figured out first part, 1250 is the lfp sample rate from session info
+
+nTimeStamps = length(aa);
+TimeStamps = (1:nTimeStamps)*fs;
+
+% Cody: this is the original code
+% ss = types.SpatialSeries('source',{'position sensor0'},...
+%     'description',{'raw sensor data from sensor 0'},...
+%     'data',aa(:,1:2),'timestamps',TimeStamps,...
+%     'starting_time',0);
+% 
+% file.acquisition.position_sensor_0 = ss;
+% 
+% 
+% ss = types.SpatialSeries('source',{'position sensor1'},...
+%     'description',{'raw sensor data from sensor 1'},...
+%     'data',aa(:,3:4),'timestamps',TimeStamps,...
+%     'starting_time',0);
+% 
+% file.acquisition.position_sensor_1 = ss;
+
+% Cody: This is the new code
+spatial_series_0 = types.core.SpatialSeries( ...
+    'data', aa(:,1:2), ...
+    'reference_frame', 'unknown', ...
+    'description', 'raw sensor data from sensor 0', ...
+    'timestamps', TimeStamps);
+
+spatial_series_1 = types.core.SpatialSeries( ...
+    'data', aa(:,3:4), ...
+    'reference_frame', 'unknown', ...
+    'description', 'raw sensor data from sensor 1', ...
+    'timestamps', TimeStamps);
+
+Position_0 = types.core.Position('SpatialSeries', spatial_series_0);
+Position_1 = types.core.Position('SpatialSeries', spatial_series_1);
+
+behavior_mod_0 = types.core.ProcessingModule( ...
+    'description',  'contains behavioral data');
+behavior_mod_1 = types.core.ProcessingModule( ...
+    'description',  'contains behavioral data');
+
+behavior_mod_0.nwbdatainterface.set(...
+    'Position', Position_0);
+behavior_mod_1.nwbdatainterface.set(...
+    'Position', Position_1);
+
+nwb.processing.set('behavior', behavior_mod_0);
+nwb.processing.set('behavior', behavior_mod_1);
```

Comment:
this overwrites the behavior module. Instead, you should put both spatial series objects into a single position object, and add that position object to a single behavior module.

Reply:
So, even for the python code, should these two position sensors be added separately as acquisitions? Or should they be combined into one acquisition? They seem separate from the other behavioral SpatialSeries that are in the processing.behavior part...